contrib/database/sql/option.go

```diff
@@ -47,3 +47,55 @@ func WithAnalyticsRate(rate float64) RegisterOption {
 		}
 	}
 }
+
+type openConfig struct {
+	serviceName   string
+	analyticsRate float64
+	dsn           string
+}
+
+// OpenOption represents an option that can be passed to Open or OpenDB.
```

Comment:
I'm fine with having multiple options structures, as it gives for better documentation, but not at the cost of prefixing the option names with `Open*`. I think that's terrible (we've already tried this in the RoundTripper and it's not one of our best). Since all of these options (except `WithDSN`) apply to both `Register` and `Open`, let's just have one global `Options` structure for the entire package. Then, `sqltrace.Option` makes a nice example of all the options of the SQL package.

It'll stay consistent with what we do in [the `net/http`](https://github.com/DataDog/dd-trace-go/blob/dda0810c69ace75c63b8f1bd64f2c300dc34effd/contrib/net/http/option.go#L20-L21) and [`google.golang.org/grpc`](https://github.com/DataDog/dd-trace-go/blob/dda0810c69ace75c63b8f1bd64f2c300dc34effd/contrib/google.golang.org/grpc/option.go#L9-L11) integrations. Plus, it's weird to technically duplicate all this code.

Reply:
Minor misunderstanding. Updated things to `Option`.